packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -191,14 +197,28 @@ class _FlutterPlatform extends PlatformPlugin {
       ));
 
       // Start the engine subprocess.
-      printTrace('test $ourTestCount: starting shell process');
+      printTrace('test $ourTestCount: starting shell process${previewDart2? " in preview-dart-2 mode":""}');
+
+      final String mainDart = previewDart2
+        ? await compile(
+            sdkRoot: artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath),
+            incrementalCompilerByteStorePath: '' /* not null is enough */,
+            mainPath: listenerFile.path,
+            packagesPath: PackageMap.globalPackagesPath
+          )
+        : listenerFile.path;
+
+      // bundlePath needs to point to a folder with `platform.dill` file.
+      final String bundlePath = previewDart2 ? artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath) : null;
+
       final Process process = await _startProcess(
```

Comment:
The --strong option needs to be passed to the engine being invoked also as we will want to also test the runtime strong mode checks.